data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateEpochCommittees.java

```diff
@@ -0,0 +1,175 @@
+import static tech.pegasys.teku.beaconrestapi.SingleQueryParameterUtils.INVALID_NUMERIC_VALUE;
+
+import io.javalin.core.util.Header;
+import io.javalin.http.Context;
+import io.javalin.http.Handler;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiParam;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import java.util.Optional;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jetbrains.annotations.NotNull;
+import tech.pegasys.teku.api.ChainDataProvider;
+import tech.pegasys.teku.api.DataProvider;
+import tech.pegasys.teku.api.response.v1.beacon.GetStateEpochCommitteesResponse;
+import tech.pegasys.teku.api.response.v1.beacon.GetStateRootResponse;
+import tech.pegasys.teku.api.schema.BeaconHead;
+import tech.pegasys.teku.beaconrestapi.handlers.AbstractHandler;
+import tech.pegasys.teku.beaconrestapi.schema.BadRequest;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.provider.JsonProvider;
+import tech.pegasys.teku.storage.client.ChainDataUnavailableException;
+import tech.pegasys.teku.util.config.Constants;
+
+public class GetStateEpochCommittees extends AbstractHandler implements Handler {
+  private static final Logger LOG = LogManager.getLogger();
+  public static final String ROUTE = "/eth/v1/beacon/states/:state_id/committees/:epoch";
+
+  private final ChainDataProvider provider;
+  private final StateValidatorsUtil stateValidatorsUtil = new StateValidatorsUtil();
+
+  public GetStateEpochCommittees(final DataProvider dataProvider, final JsonProvider jsonProvider) {
+    super(jsonProvider);
+    this.provider = dataProvider.getChainDataProvider();
+  }
+
+  GetStateEpochCommittees(
+      final ChainDataProvider chainDataProvider, final JsonProvider jsonProvider) {
+    super(jsonProvider);
+    this.provider = chainDataProvider;
+  }
+
+  @OpenApi(
+      path = ROUTE,
+      method = HttpMethod.GET,
+      summary = "Get all committees for epoch",
+      tags = {TAG_V1_BEACON},
+      description = "Retrieves the committees for the given state at the given epoch.",
+      pathParams = {
+        @OpenApiParam(name = PARAM_STATE_ID, description = PARAM_STATE_ID_DESCRIPTION),
+        @OpenApiParam(name = PARAM_EPOCH, description = PARAM_EPOCH_DESCRIPTION)
+      },
+      queryParams = {
+        @OpenApiParam(
+            name = COMMITTEE_INDEX,
+            description = COMMITTEE_INDEX_QUERY_DESCRIPTION,
+            isRepeatable = false),
+        @OpenApiParam(name = SLOT, description = SLOT_QUERY_DESCRIPTION, isRepeatable = false)
+      },
+      responses = {
+        @OpenApiResponse(
+            status = RES_OK,
+            content = @OpenApiContent(from = GetStateRootResponse.class)),
```

Comment:
needs a different response object here or openapi will be incorrect in the schema tag

Reply:
Done.